### PR TITLE
Script function support

### PR DESCRIPTION
**Pull request by fedddddd**

### CODE REVIEW

File: src/client/game/scripting/execution.cpp [hunk 1]
```diff
@@ -111,6 +111,24 @@ namespace scripting
...
+
+		stack_isolation _;
+		for (auto i = arguments.rbegin(); i != arguments.rend(); ++i)
+		{
+			scripting::push_value(*i);
+		}
+
+		game::AddRefToObject(id);
+
+		const auto local_id = game::AllocThread(id);
```
Comment (momo5502) [inline src/client/game/scripting/execution.cpp:126]: Don't you have to free the thread again?

**Review by momo5502 — CHANGES_REQUESTED** [review]

File: src/client/game/scripting/execution.cpp [hunk 2]
```diff
@@ -125,6 +125,7 @@ namespace scripting
 
 		const auto local_id = game::AllocThread(id);
 		const auto result = game::VM_Execute(local_id, pos, (unsigned int)arguments.size());
+		game::RemoveRefToObject(result);
```
Comment (momo5502) [inline src/client/game/scripting/execution.cpp:128]: Don't you have to use Scr_FreeThread or something like that? Or is that something different

Comment (mjkzy) [reply src/client/game/scripting/execution.cpp:128]: if he's hooking/messing with something to do with executing threads (or Scr_ExecThread), then I believe so

Comment (fedddddd) [reply src/client/game/scripting/execution.cpp:128]: Scr_FreeThead simply calls RemoveRefToObject so it makes no difference

**Comment by momo5502** [comment]
I merged it, but still. Can VM_Execute throw exceptions? I think it can. Catching these might maybe be potentially important? Or does it handle all exceptions internally?

**Comment by fedddddd** [comment]
It handles them internally

**Comment by momo5502** [comment]
Perfect

**Pull request merged**